libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1589,7 +1626,7 @@ SMTEncoder::VariableIndices SMTEncoder::copyVariableIndices()
 void SMTEncoder::resetVariableIndices(VariableIndices const& _indices)
 {
 	for (auto const& var: _indices)
-		m_context.variable(*var.first)->index() = var.second;
+		m_context.variable(*var.first)->setIndex(var.second);
```

Comment:
@ekpyron this was the problem on the BMC side (which uses the `mergeVariables` function).
The index of the symbolic array was being reset after a branch, but not of the tuple.

Reply:
The idea here is that we need to set the current index but keep the `next` as it is, without zeroing, because we still need to `ite` (merge) after the branches. That's why I added the function `SSAVariable::setIndex` that does that. The old `SSAVariable::reset` zeroes the current index and sets next to 1, effectively resetting.

Reply:
Hm - I'm not sure I understand this entirely, but I'd go on a leap and trust you that this works on this one :-).